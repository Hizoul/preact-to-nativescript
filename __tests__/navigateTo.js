import { Preact, navigateTo, goBack } from "../index"
import * as frame from "tns-core-modules/ui/frame"

test("navigateTo", () => {
  const topmost = frame.topmost()
  console.log("got topmost", topmost)
  expect(topmost.navigatedTo).toMatchSnapshot("didnt navigate yet")
  expect(topmost.backCallAmount).toBe(0)
  navigateTo(Preact.h("page"))
  expect(topmost.navigatedTo).toMatchSnapshot("navigated to page")
  expect(topmost.backCallAmount).toBe(0)
  topmost.goBack()
  expect(topmost.backCallAmount).toBe(1)
  goBack()
  expect(topmost.backCallAmount).toBe(2)
})
