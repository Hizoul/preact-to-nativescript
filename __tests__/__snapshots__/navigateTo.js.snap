// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigateTo: didnt navigate yet 1`] = `"nowhere yet"`;

exports[`navigateTo: navigated to page 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {},
  "_eventListener": Array [],
  "attributes": Array [],
  "childNodes": Array [],
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "_removeView": [Function],
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [
      [Circular],
    ],
    "mergeInto": undefined,
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [],
      "key": undefined,
      "nodeName": "page",
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [],
    "key": undefined,
    "nodeName": "page",
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;
